x/concentrated-liquidity/position.go

```diff
@@ -62,6 +63,37 @@ func (k Keeper) initOrUpdatePosition(
 
 	// TODO: consider deleting position if liquidity becomes zero
 
+	// Create records for relevant uptime accumulators here.
+	uptimeAccumulators, err := k.getUptimeAccumulators(ctx, poolId)
+	if err != nil {
+		return err
+	}
+
+	for uptimeId, uptime := range types.SupportedUptimes {
+		if position.FrozenUntil.Sub(ctx.BlockTime()) >= uptime {
```

Comment:
Can we have some tests around this check? I don't think we have any tests where we manipulate the block time

Reply:
non-blocking: It might even be helpful to encapsulate all record creation logic in a separate function and test it

Reply:
For your first point, this could be useful to add but every test setup in `position_test.go` uses a different time (since it sets up to local time) so I believe tests for different uptimes are already tested